### DISCUSSION TITLE
Consider starting VS Code from within an Anaconda terminal session

### DISCUSSION THREAD

**Discussion by greazer** [microsoft/vscode-jupyter#9399] [Tips, Tricks and Workarounds]
While both the Jupyter and Python extensions strive to fully support the Anaconda "out of the box", there are bugs that can be experienced in each of the extensions as well as in some of the Anaconda machinery itself. 

One way of reducing the surface area for problems you may encounter is to run VS Code from within the context of an Anaconda environment to begin with. This is typically how other notebook tools like Jupyter are normally run, so it's not really that unusual of a thing to do.

You can do this from either Anaconda navigator:

<img width="1758" alt="image" src="https://user-images.githubusercontent.com/11685408/158714400-3344b1da-3b1a-4d3d-828b-e529b7fd214d.png">

or from an Anaconda prompt:

<img width="386" alt="image" src="https://user-images.githubusercontent.com/11685408/158714575-c80c0be9-7c12-4234-b312-ab3d4458ad34.png">

IMPORTANT NOTE: Currently the Anaconda environment that was active when you ran VS Code is not automatically selected as your Python interpreter or notebook kernel. So make sure to choose the specific environment you want to use by running either the ```Python: Select Interpreter``` or ```Notebook: Select Notebook Kernel``` command.

<img width="566" alt="image" src="https://user-images.githubusercontent.com/11685408/158715158-11552a4b-825c-4a89-8dfb-d7ea69ac43d9.png">

<img width="551" alt="image" src="https://user-images.githubusercontent.com/11685408/158715212-73db6817-254a-4a51-a296-ad9b5f99c515.png">

**Comment by DonJayamanne** [comment]
> urrently the Anaconda environment that was active when you ran VS Code is not automatically s


@greazer this is something I suggested in the past, we can detect the activated conda environment via environment variables and promote that kernel as the preferred kernel. Moved that discussion to this issue https://github.com/microsoft/vscode-jupyter/issues/9115
